components/dashboard/src/settings/SelectIDEModal.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import SelectIDE from "./SelectIDE";
+import Modal from "../components/Modal";
+
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
+
+export interface SelectEditorModal {
```

Comment:
Go with a close function so that other components can use it like `await modal`

Reply:
And will be useful in this issue https://github.com/gitpod-io/gitpod/issues/9534